workbench-core/accounts/src/services/projectService.ts

```diff
@@ -254,6 +255,55 @@ export default class ProjectService {
     return newProject;
   }
 
+  /**
+   * Update the name or description of an existing project.
+   *
+   * @param request - a {@link UpdateProjectRequest} object that contains the id of the project to update
+   *                  as well as the new field values
+   * @returns a {@link Project} object that reflects the changes requested
+   */
+  public async updateProject(request: UpdateProjectRequest): Promise<Project> {
+    const { projectId, updatedValues } = request;
+
+    // disregard empty strings as valid input
+    if (updatedValues.name !== undefined && _.isEmpty(updatedValues.name)) {
+      updatedValues.name = undefined;
+    }
+    if (updatedValues.description !== undefined && _.isEmpty(updatedValues.description)) {
+      updatedValues.description = undefined;
+    }
+
+    // verify at least one attribute is being updated
+    if (!updatedValues.name && !updatedValues.description) {
+      throw Boom.badRequest('You must supply a new nonempty name and/or description to update the project.');
+    }
+
+    // if updating name, verify it is not in use and project exists
+    if (updatedValues.name) {
+      await Promise.all([this._isProjectNameInUse(updatedValues.name), this.getProject({ projectId })]);
+    } else {
+      // else, verify project still exists
+      await this.getProject({ projectId });
+    }
+
+    // update project DDB item
+    try {
+      const updateResponse = await this._aws.helpers.ddb.updateExecuteAndFormat({
```

Comment:
the dynamo db service should be passed into through the constructor so that it is clear `ProjectService` depends on the dynamo db service. Accessing it through `this._aws.helpers` is an anti-pattern because it hides the relationship and also makes this class vulnerable to any changes in the `AwsService` class and `Helpers` class

Reply:
This is how this service is currently interacting with DB Service. I hear what you are saying, but again, I think this is an opportunity for a room for improvement item for the entire file and not just this method. The refactor can come later if we decide to make our services accept a db service param.

Reply:
Going to refactor for all of project service before merging to develop